src/Controller/StrawberryfieldFlavorDatasourceSearchController.php

```diff
@@ -129,6 +134,39 @@ public function search(Request $request, ContentEntityInterface $node, string $f
       return $response;
 
     }
+    elseif ((($format == 'miniocr') || ($format == 'djvuxml')) && (is_numeric($page))) {
+      $indexes = StrawberryfieldFlavorDatasource::getValidIndexes();
+      $miniocr = $this->miniocrfromSolrIndex(
+        $node->id(),
+        $processor,
+        $fileuuid,
+        $indexes,
+        $page
+      );
+
+      if ($format == 'miniocr') {
+        $output = $miniocr;
+      }
+      else {
+        $output = $this->miniocr2djvuxml($miniocr);
+      }
+
+      $response = new CacheableResponse(
+        $output,
+        200,
+        ['content-type' => 'application/xml'],
+        TRUE
+      );
+      if ($response) {
+        // Set CORS. IIIF and others will assume this is true.
+        $response->headers->set('access-control-allow-origin', '*');
+        $response->addCacheableDependency($node);
```

Comment:
I need to check if we can add a DataSource ID as cacheable response too. That way if the Page itself changes/is added we get the right way. Also we should not cache empty outputs or not found pages